lib/subnet-routes.js

```diff
@@ -0,0 +1,79 @@
+/** @import { RequestWithSubnet } from './typings.js' */
+/** @import { FastifyInstance, FastifyReply } from 'fastify' */
+import { subnetPreHandlerHook } from './request-helpers.js'
+
+/**
+ * Define the subnet routes
+ * @param {FastifyInstance} app
+ */
+export const subnetRoutes = (app) => {
+  app.addHook('preHandler', subnetPreHandlerHook)
+
+  app.post(
+    '/:subnet/measurement',
+    {
+      schema: {
+        body: {
+          type: 'boolean'
+        }
+      }
+    },
+    /**
+     * @param {RequestWithSubnet} request
+     * @param {FastifyReply} reply
+     */
+    async (request, reply) => {
+      const client = await app.pg.connect()
+      try {
+        await client.query(
+          'UPDATE measurements SET total = total + 1, successful = successful + $1 WHERE subnet = $2',
+          [request.body ? 1 : 0, request.params.subnet]
+        )
+      } finally {
+        client.release()
+      }
+    }
+  )
+
+  app.get(
+    '/:subnet',
+    {
+      schema: {
+        response: {
+          200: {
+            type: 'object',
+            properties: {
+              total: {
+                type: 'string',
+                format: 'int64',
+                pattern: '^[0-9]*$'
+              },
+              successful: {
+                type: 'string',
+                format: 'int64',
+                pattern: '^[0-9]*$'
+              }
+            }
```

Comment:
In JSON Schema, all properties are optional by default. That's great for extensibility/forward-compatibility, but it makes it more difficult to consume such data in strictly typed languages - clients have to check whether `total` and `successful` were provided by the server. In our case, we want the server to always provide both properties, so let's describe that in the schema too.

```suggestion
            },
            required: ['total', 'successful']
```

Reply:
I wasn't aware of that, thank you for pointing it out 🙏🏻